### PR TITLE
backend "list all agencies" route, controller

### PR DESCRIPTION
**Pull request by nigel0511**
#1405 
I have implemented a route to request list of all agencies from database, please comment.

### CODE REVIEW

File: server/src/modules/agency/agency.routes.ts [hunk 1]
```diff
@@ -13,18 +13,14 @@ export const routeAgencies = ({
...
    * @return 404 if agency is not found
    * @return 500 if database error
    * @access Public
    */
-  router.get(
-    '/',
-    [query('shortname').optional(), query('longname').optional()],
-    controller.getSingleAgency,
-  )
+  router.get('/all', controller.listAllAgencies)
```
Comment (LoneRifle) [inline server/src/modules/agency/agency.routes.ts]:
```suggestion
  router.get('/', controller.listAllAgencies)
```

File: server/src/modules/agency/agency.routes.ts [hunk 2]
```diff
@@ -13,18 +13,14 @@ export const routeAgencies = ({
   const router = express.Router()
 
   /**
-   * Find an agency by their shortname or longname
-   * @route  GET /api/agencies?<params>
+   * List all agencies
+   * @route  GET /api/agencies/all
```
Comment (LoneRifle) [inline server/src/modules/agency/agency.routes.ts]:
```suggestion
   * @route  GET /api/agencies
```

**Review by LoneRifle — CHANGES_REQUESTED** [review]
lgtm otherwise for now.

**Comment by LoneRifle** [comment]
@siuting0511 - I also want to see some tests that verify your implementation. Could you help us with that please?

**Comment by nigel0511** [comment]
Specifically, what test would you like me to test the code with?

**Comment by LoneRifle** [comment]
> Specifically, what test would you like me to test the code with?

I would like some comprehensive unit test coverage in `agency.service.spec` and `agency.controller.spec` that verify that the implementations work as expected in both regular and error conditions, as well as integration test coverage in `agency.routes.spec` that ensures that both controller and service work when put together. You'll find this under `__tests__`.

I'll approve and run CI for your PR now, and you can see how your tests run.

**Comment by nigel0511** [comment]
ok, I will work on that.

**Comment by nigel0511** [comment]
I have added unit test for the backend agency route, controller and server, it seems like they are working well, please ignore the modified frontend code for now as they are not completed.

**Review by LoneRifle — APPROVED** [review]
I've reworked and force-pushed your branch so that it incorporates the backend changes that are of interest. As part of this, I've dropped the frontend changes that you said were incomplete and a work-in-progress, as I wanted to keep this PR focused on the backend changes.

A couple of points for you that I want to bring up:
- Use the latest version of Node and npm for this codebase. You might have noticed that your commit included a bunch of changes in the package lockfile as a result of you running `npm install` using an older version of npm that does not support v2 of the lockfile spec. Using a more recent version of npm would have avoided that.

- Pay close attention to, and _do not commit_ merge conflicts! These would cause your codebase to be unbuildable as git would litter your files with indicators where merge conflicts occur, expecting you to resolve them, not commit and push them into the repo.

If you wish to recover the changes you made to the frontend for a later PR, the relevant commit is e696f46.

**Comment by nigel0511** [comment]
Thank you for pointing them out! I will definitely keep those in mind in the future.

**Pull request merged**